### PR TITLE
Fixes guardian summoners not dusting properly

### LINKED ISSUE #47238: Cremating a holoparasite kills the host but doesn't dust them
No description needed

Issue reported from  Round ID: 121813 (/tg/Station Sybil [ENGLISH] [US-WEST] [100% LAG FREE])

Reporting client version: 512.1488

### PR DESCRIPTION
**Pull request by Neotwr**
## About The Pull Request
Fixes #47238 

## Changelog
:cl:
fix: Holoparasite summoners now properly dust when their guardians are cremated
/:cl:

### CODE REVIEW

**Comment by nemvar** [comment]
Remove the dusting code from Life() and move that here.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 1]
```diff
@@ -204,7 +187,24 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
...
+		if(summoner.stat == DEAD)
+			forceMove(summoner.loc)
+			to_chat(src, "<span class='danger'>Your summoner has died!</span>")
+			visible_message("<span class='danger'><B>\The [src] dies along with its user!</B></span>")
+			summoner.visible_message("<span class='danger'><B>[summoner]'s body is completely consumed by the strain of sustaining [src]!</B></span>")
+			for(var/obj/item/W in summoner)
+				if(!summoner.dropItemToGround(W))
+					qdel(W)
+			summoner.dust()
+			death(TRUE)
```
Comment (YPOQ) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: This looks like infinite recursion

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 2]
```diff
@@ -204,7 +187,24 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
...
+			qdel(src)
+	else
+		to_chat(src, "<span class='danger'>Your summoner has died!</span>")
+		visible_message("<span class='danger'><B>[src] dies along with its user!</B></span>")
+		death(TRUE)
```
Comment (YPOQ) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: .

**Review by YPOQ — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/guardian/guardian.dm (hunk 1 above)
Comment (Neo-0) [reply code/modules/mob/living/simple_animal/guardian/guardian.dm]: no, it works

File: code/modules/mob/living/simple_animal/guardian/guardian.dm (hunk 2 above)
Comment (AnturK) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: This is just recursively calling itself without any escape ? Same above.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 3]
```diff
@@ -204,7 +187,24 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 	..()
 	if(summoner)
 		to_chat(summoner, "<span class='danger'><B>Your [name] died somehow!</span></B>")
-		summoner.death()
+		summoner.dust()
+	if(!QDELETED(summoner))
+		if(summoner.stat == DEAD)
+			forceMove(summoner.loc)
+			to_chat(src, "<span class='danger'>Your summoner has died!</span>")
+			visible_message("<span class='danger'><B>\The [src] dies along with its user!</B></span>")
```
Comment (AnturK) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: These don't really make sense if the death was initiated by the guardian.

**Review by AnturK — CHANGES_REQUESTED** [review]
Please explain how is this supposed to work because it doesn't make any sense right now.

**Comment by Neo-0** [comment]
Initially this PR was made to fix this issue #47238 
I'll leave it the way it was, fixing the issue
I'm not a coder

**Pull request merged**